setup db to postgres

SHoward-A opened the pull request:
setup docker-compose for dev/test
setup static file storage to s3 for prod, but testing remaining.

Commit:
setup db to postgres
setup docker-compose for dev/test
setup static file storage to s3 for prod, but testing remaining.

khalil-omer commented on docker/test-server/etebase-server.ini:
```diff
@@ -3,10 +3,19 @@ secret_file = secret.txt
 debug = true
 static_root = /data/static
 media_root = /data/media
+AWS_ACCESS_KEY_ID = AKIARXDELOGPTJKCHC5B
```
Should never hardcode credentials and commit that. Always should use env variables. https://www.twilio.com/blog/environment-variables-python

SHoward-A replied:
Actually in this repo, .ini file is used to store all environment variables, if you see in settings.py file, you will notice that all the env variables are loaded from this config file (.ini)

khalil-omer commented on etebase_server/settings.py:
```diff
@@ -158,6 +158,14 @@
+        AWS_DEFAULT_ACL = 'public-read'
+
     if "redis_uri" in section:
         ETEBASE_REDIS_URI = section.get("redis_uri")
```
@vitaly512 I guess we should configure Redis as well for both dev and prod.

SHoward-A replied:
Noted.

khalil-omer commented on docker/test-server/etebase-server.ini:
```diff
@@ -3,10 +3,19 @@ secret_file = secret.txt
+ENGINE = django.db.backends.postgresql_psycopg2
+NAME = eteserverdb
+USER = root
+PASSWORD = mysecretpassword
```
Things like db user and password should be env variables

SHoward-A replied:
Again, this file is used to store env variables in this repo, if you want to use .env file, then I can add/replace it with that as well. 
What you say?

khalil-omer replied:
@vitaly512 yes we should update to use env vars for all these secrets.

khalil-omer replied:
You can create a .env.example file to show what envs are needed. You can also create your own README file.

SHoward-A replied:
Okay, I will update that.

khalil-omer commented on docker/test-server/Dockerfile:
```diff
@@ -6,12 +6,12 @@ ENV PIP_DISABLE_PIP_VERSION_CHECK=1
 
 # install packages and pip requirements first, in a single step,
-COPY /requirements.txt /requirements.txt
+COPY /requirements.in/base.txt /requirements.txt
```
I don't understand this change. Can you explain?

SHoward-A replied:
This is base requirement file (located in this directory: **/requirements.in/**), I just use that one, because /requirements.txt (in root of repo) is autogenerated file that's why I avoid that.

khalil-omer commented on docker/test-server/Dockerfile:
```diff
@@ -6,12 +6,12 @@ ENV PIP_DISABLE_PIP_VERSION_CHECK=1
     pip install -U pip ;\
     pip install --no-cache-dir --progress-bar off -r /requirements.txt ;\
-    apk del .build-deps make gcc coreutils ;\
+    # apk del .build-deps make gcc coreutils ;\  # do not uncomment this, otherwise errors start producing.
```
Why should we comment this line? If it is from the original repo, it is probably an important step.

SHoward-A replied:
deleting these build-deps files causing docker to failed, it was showing error of missing libraries (and those libs were included in the docker file), so there was no point to run docker without commenting this line.

khalil-omer replied:
@vitaly512 maybe commenting out this build deps is related to the failure we are having?

SHoward-A replied:
You mean media files failure to AWS S3? I don't think so, it has to do any thing with that.

khalil-omer commented on docker/test-server/Dockerfile:
```diff
@@ -21,18 +21,19 @@ RUN set -ex ;\
-    cd /app ;\
-    python manage.py migrate ;\
-    python manage.py collectstatic --noinput
+    cd /app # ;\
```
Can you explain this change?

SHoward-A replied:
I have created a new file **entrypoint.sh**, and moved these lines to there, as while using docker-compose, we should execute these lines only when container starts.

khalil-omer commented on docker/test-server/Dockerfile:
```diff
@@ -21,18 +21,19 @@ RUN set -ex ;\
 
 ENV ETESYNC_VERSION=${ETESYNC_VERSION}
-VOLUME /data
+#VOLUME /data
```
Why comment out the volume? It needs to store files. In prod we will use s3, but in local we would use local volume.

SHoward-A replied:
Volume line is moved to docker-compose file, if we use docker-compose we mount volume through docker-compose file.

Commit:
ini file updated

Commit:
- Updates/improvements made in docker files.
- Added docker-compose implementation for prod environment.
- New python packages added in requirements file.

Commit:
eteserver updates.

Commit:
- redis added in docker-compose.
- save only media files to aws s3, save static files on local

khalil-omer commented on docker/prod-server/etebase-server.ini:
```diff
@@ -3,6 +3,9 @@ secret_file = secret.txt
 media_root = ./data/media
+static_url = /static/
+media_url = /user-media/
+redis_uri = redis://localhost:6379
```
@vitaly512 in prod, redis url should be inserted through an environment variable

Commit:
- custom user verification feature added during signup process.
- requirements, settings, and configurations updates.

Commit:
- settings updates.

khalil-omer commented on docker/prod-server/etebase-server.ini:
```diff
@@ -0,0 +1,27 @@
+ENGINE = django.db.backends.postgresql_psycopg2
+NAME = eteserverdb
+USER = root
+PASSWORD = mysecretpassword
```
@vitaly512 these database values should pull from env variables, not be hardcoded.

SHoward-A replied:
Okay, I will add .env file for this.

Commit:
- ete-server updates.
- .env file added for environment variables.

SHoward-A committed:
Update collection.py

add logs

Commit:
fix s3 absolute path error

Commit:
remove commited keys
